Mathlib/Topology/Algebra/ProperAction.lean

```diff
@@ -0,0 +1,342 @@
+/-
+Copyright (c) 2024 Anatole Dedeker. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Anatole Dedeker, Etienne Marion, Florestan Martin-Baillon, Vincent Guirardel
+-/
+
+import Mathlib.Topology.ProperMap
+import Mathlib.GroupTheory.Subgroup.Actions
+import Mathlib.Topology.Algebra.MulAction
+import Mathlib.Topology.Defs.Sequences
+import Mathlib.Topology.Sequences
+
+/-!
+# Proper Action
+
+In this file we define proper action of a group on a topological space, and we prove that in this
+case the quotient space is T2. We also give equivalent definitions of proper action using
+ultrafilters and show the transfer of proper action to a closed subgroup.
+
+## Main definitions
+
+* `ProperSMul` : a group `G` acts properly on a topological space `X`
+  if the map `(g, x) ↦ (g • x, x)` is proper, in the sense defined in `IsProperMap`.
+
+## Main statements
+
+* `t2Space_of_ProperSMul`: If a group `G` acts properly on a topological space `X`,
+  then the quotient space is Hausdorff (T2).
+* `t2Space_of_properSMul_of_t2Group`: If a T2 group acts properly on a topological space,
+  then this topological space is T2.
+
+## Notation
+
+
+
+## Implementation details
+
+
+
+## References
+
+* [F. Bar, *Quuxes*][bibkey]
+
+## Tags
+
+Foobars, barfoos
+-/
+
+
+
+
+/-- Additive version of proper action in the sense of Bourbaki:
+the map `G×X→ X×X` is a proper map `isProperMap`
+-/
+@[mk_iff]
+class ProperVAdd (G X : Type*) [TopologicalSpace G] [TopologicalSpace X] [AddGroup G]
+    [AddAction G X] : Prop where
+  /-- Additive version of proper action in the sense of Bourbaki:
+the map `G×X→ X×X` is a proper map `isProperMap`  -/
+  isProperMap_vadd_pair' : IsProperMap (fun gx ↦ ⟨gx.1 +ᵥ gx.2, gx.2⟩ : G × X → X × X)
+
+/-- Proper action in the sense of Bourbaki:
+the map `G×X→ X×X` is a proper map `isProperMap`
+-/
+@[to_additive existing, mk_iff]
+class ProperSMul (G X : Type*) [TopologicalSpace G] [TopologicalSpace X] [Group G]
+    [MulAction G X] : Prop where
+  isProperMap_smul_pair' : IsProperMap (fun gx ↦ ⟨gx.1 • gx.2, gx.2⟩ : G × X → X × X)
+
+attribute [to_additive existing] properSMul_iff
+
+/-- By definition, if G acts properly on X
+the map `G×X→ X×X` is a proper map `isProperMap`
+-/
+@[to_additive]
+lemma isProperMap_smul_pair (G X : Type*) [Group G] [MulAction G X]
+    [TopologicalSpace G] [TopologicalSpace X] [ProperSMul G X] :
+    IsProperMap (fun gx ↦ ⟨gx.1 • gx.2, gx.2⟩ : G × X → X × X) :=
+  ProperSMul.isProperMap_smul_pair'
+
+open Filter Topology
```

Comment:
Could you move this at the top of the file (right after the module docstring) and also open `Set`? Then you can remove basically all `Filter.` and `Set.` in the rest of the file (some may still be needed for disambiguation).